src/index.ts

```diff
@@ -27,15 +25,21 @@ export interface Options {
 /**
  * Returns the base URL while taking into account the GCE_METADATA_IP
  * environment variable if it exists.
- * By using a function instead of a const, testing this functionality is made
- * easier.
  *
- * @returns The base URL, eg. http://169.254.169.254/computeMetadata/v1.
+ * @returns The base URL, e.g., http://169.254.169.254/computeMetadata/v1.
  */
-function getBaseUrl(): string {
-  return process.env.GCE_METADATA_IP
-    ? `http://${process.env.GCE_METADATA_IP}${BASE_PATH}`
-    : BASE_URL;
+function getBaseUrl(baseUrl?: string) {
+  const metadataBasePath = '/computeMetadata/v1';
```

Comment:
Since this doesn't change, maybe we should keep it a global `BASE_PATH` constant?